src/org/omegat/core/team2/impl/GITRemoteRepository2.java

```diff
@@ -382,6 +380,34 @@ public static boolean deleteDirectory(File path) {
         return (path.delete());
     }
 
+    /**
+     * Retrieve default branch name from remote origin HEAD.
+     * @param repositoryUrl Remote repository URL
+     * @return default branch name, ordinary "main"(recent popular) or "master"(old default)
+     */
+    public static String getDefaultBranchName(final String repositoryUrl) {
+        try {
+            Map<String, Ref> gitMap = Git.lsRemoteRepository().setRemote(repositoryUrl).callAsMap();
+            Ref head = gitMap.get("HEAD");
+            if (head.isSymbolic()) {
+                return getBranchFromRef(head.getTarget().getName());
+            } else {
```

Comment:
If you follow my advice and add a null-check-early-return, it might make things more readable to take the `else` code out of the `else` branch and just put it flat after the `if`, since the `if` returns.